docs/content/sips/000-sqlite.md

```diff
@@ -0,0 +1,139 @@
+
+### What about `wasi-sql`?
+
+[`wasi-sql`](https://github.com/WebAssembly/wasi-sql) is a work-in-progress spec for a generic SQL interface that aims to support "the features commonly used by 80% of user application". It is likely that when `wasi-sql` is more mature users will be able to successfully use functionality based on the `wasi-sql` interface to interact with a sqlite databases. However, there are still reasons that a dedicated sqlite interface would still be useful:
+
+* For the 20% of use cases where `wasi-sql` is too generic, a dedicated `sqlite` interface can provide that functionality. 
+* The `wasi-sql` spec is under active investigation, and there are large questions about how to best support such a wide breadth of sql flavors. This implementation can help clarify those questions and push upstream work further along.
+
+## Proposal
+
+In order to support sqlite, the following need to be added to Spin:
+
+- A `WIT` file that defines the sqlite interface
+- SDK implementations for various programming languages
+- A default local sqlite store (note: Spin already uses sqlite for the KV implementation so this should be trivial)
+- Potentially runtime configuration for configuring how sqlite is provisioned.
+- Potentially a mechansim for handling database migrations
+
+### Interface (`.wit`)
+
+We will start with the `wasi-sql` interface but deliberately change that interface as to better match sqlite semantics. This will ensure that we're not simply implementing early versions of the `wasi-sql` interface while still having good answers for why the interface differs when it does.
+
+Like `wasi-sql` and the key-value store, we model resources such as database connections as pseudo-[resource handles](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md#item-resource) which may be created using an `open` function and disposed using a `close` function. Each operation on a connection is a function which accepts a handle as its first parameter.
+
+Note that the syntax of the following `WIT` file matches the `wit-bindgen` version currently used by Spin, which is out-of-date with respect to the latest [`WIT` specification](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md) and implementation. Once we're able to update `wit-bindgen`, we'll update the syntax of all the Spin `WIT` files, including this one.
+
+```fsharp
+// A handle to an open sqlite instance
+type connection = u32
+
+// The set of errors which may be raised by functions in this interface
+variant error {
+  // The host does not recognize the database name requested.
```

Comment:
Well it's not that the name is defined, it's that there is no database by that name. This differs from `access-denied` in that in the `access-denied` case there is explicitly no access granted for a database with that name while in the `no-such-database` case it's not a question of permission, there simply is no database that exists for that identifier. Any suggestions on how to make that clearer?

Reply:
"No database named {...} is defined." or "There is no database named {...}." maybe?  (Of these, I somewhat prefer the first, but I am sure there are better options.)

Your comment also made me wonder about which of "not defined" and "not granted access" takes precedence.  If I write a component that tries to use a database called `no-ivans`, but the manifest does not grant my component access to `no-ivans`, will my component get `access-denied` rather than "leaking" information about whether the database exists or doesn't exist?

(Longer term this should be a non-issue because `spin up` should fail manifest validation if it refers to a name that isn't defined, but I don't think we can do that today.)